### PR TITLE
Add tests for rocket_clean_cache_dir function

### PR DESCRIPTION
**Pull request by engahmeds3ed**
- [x] Add Unit and Integration tests for rocket_clean_cache_dir method which locates in inc/functions/files.php

- [x] Refactor the code itself to use SPL iterators instead of glob function.

  Why?
- To be able to make the code testable.
- To make the code faster as this function is called multiple times through the code.

### CODE REVIEW

File: inc/functions/files.php [hunk 1]
```diff
@@ -979,11 +979,14 @@ function rocket_clean_cache_dir() {
...
 
-	// Delete all caching files.
-	$dirs = glob( WP_ROCKET_CACHE_PATH . '*', GLOB_NOSORT );
-	if ( $dirs ) {
-		foreach ( $dirs as $dir ) {
-			rocket_rrmdir( $dir );
+	$filesystem = rocket_direct_filesystem();
+
+	// Delete all caching files/folders.
+	foreach ( _rocket_get_cache_dirs( '.' ) as $path ) {
```
Comment (Tabrisrp) [inline inc/functions/files.php:985]: `_rocket_get_cache_dirs()` expects an URL host as the first parameter, passing it a dot like this is strange.

How about using this instead?

```suggestion
	foreach ( _rocket_get_cache_path_iterator( _rocket_get_wp_rocket_cache_path() ) as $path ) {
```

File: inc/functions/files.php [hunk 2]
```diff
@@ -979,11 +979,14 @@ function rocket_clean_cache_dir() {
...
+		if ( ! $filesystem->is_dir( $path ) ) {
+			$filesystem->delete( $path );
+		}else {
+			rocket_rrmdir( $path );
```
Comment (Tabrisrp) [inline inc/functions/files.php:989]:
```suggestion
		if ( ! $filesystem->is_dir( $path ) ) {
			$filesystem->delete( $path );
			continue;
		}

		rocket_rrmdir( $path );
```

Comment (engahmeds3ed) [reply inc/functions/files.php:989]: Really inside loops I don't like using continue and break as I think they are `hidden traps` and makes code not readable, what do you think?

Comment (hellofromtonya) [reply inc/functions/files.php:989]: The same can be said of `else`, as some think `else` blocks are evil.

A good rule of thumb is this: 

- If no other processing should happen after the `if` block, then bail out: `return`, `continue`, or `break`.

Why?  Nothing else should happen after this `if` is done. Therefore, adding a bail out makes it very clear that the work is done.

Comment (engahmeds3ed) [reply inc/functions/files.php:989]: Great, nice Idea.
Thanks @Tabrisrp @hellofromtonya .

**Comment by iCaspar** [comment]
This PR was ported to [module-filesystem, PR #3](https://github.com/wp-media/module-filesystem/pull/3)

**Pull request closed without merging**